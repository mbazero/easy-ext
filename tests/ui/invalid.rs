mod basic {
    use easy_ext::ext;

    #[ext(NoValueConst)]
    impl str {
        const ASSOC: u8; //~ ERROR expected `=`
    }
    #[ext(NoValueTy)]
    impl str {
        type Assoc; //~ ERROR expected `=`
    }
    #[ext(NoValueFn)]
    impl str {
        fn assoc(); //~ ERROR expected curly braces
    }

    #[ext(Macro)]
    impl str {
        mac!(); //~ ERROR expected one of: `default`, `fn`, `const`, `type`
    }
}

mod visibility {
    use easy_ext::ext;

    #[ext(AssocLevel1)]
    impl str {
        pub const ASSOC1: u8 = 1;
        const ASSOC2: u8 = 2; //~ ERROR all associated items must have a visibility of `pub`
    }

    #[ext(AssocLevel2)]
    impl str {
        fn assoc1(&self) {}

        pub fn assoc2(&self) {} //~ ERROR all associated items must have inherited visibility
    }

    #[ext(AssocLevel3)]
    impl str {
        pub(crate) type Assoc1 = ();
        pub type Assoc2 = (); //~ ERROR all associated items must have a visibility of `pub(crate)`
    }

    #[ext(pub ImplLevel1)]
    impl str {
        fn assoc1(&self) {}

        pub fn assoc2(&self) {} //~ ERROR all associated items must have inherited visibility
    }
}

fn main() {}
