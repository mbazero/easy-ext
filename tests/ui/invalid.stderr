error: expected `=`
 --> tests/ui/invalid.rs:8:24
  |
8 |         const ASSOC: u8; //~ ERROR expected `=`
  |                        ^

error: expected `=`
  --> tests/ui/invalid.rs:12:19
   |
12 |         type Assoc; //~ ERROR expected `=`
   |                   ^

error: expected `{`
  --> tests/ui/invalid.rs:16:19
   |
16 |         fn assoc(); //~ ERROR expected `{`
   |                   ^

error: expected one of: `default`, `fn`, `const`, `type`
  --> tests/ui/invalid.rs:21:9
   |
21 |         mac!(); //~ ERROR expected one of: `default`, `fn`, `const`, `type`
   |         ^^^

error: unexpected token: `,`
  --> tests/ui/invalid.rs:25:19
   |
25 |     #[ext(ExtraArg,)] //~ ERROR unexpected token
   |                   ^

error: use `pub impl` instead
  --> tests/ui/invalid.rs:28:11
   |
28 |     #[ext(pub OldVisSyntax1)] //~ ERROR use `pub impl` instead
   |           ^^^

error: use `pub(crate) impl` instead
  --> tests/ui/invalid.rs:31:11
   |
31 |     #[ext(pub(crate) OldVisSyntax2)] //~ ERROR use `pub(crate) impl` instead
   |           ^^^^^^^^^^

error: all associated items must have a visibility of `pub`
  --> tests/ui/invalid.rs:41:15
   |
41 |         const ASSOC2: u8 = 2; //~ ERROR all associated items must have a visibility of `pub`
   |               ^^^^^^

error: all associated items must have inherited visibility
  --> tests/ui/invalid.rs:48:9
   |
48 |         pub fn assoc2(&self) {} //~ ERROR all associated items must have inherited visibility
   |         ^^^

error: all associated items must have a visibility of `pub(crate)`
  --> tests/ui/invalid.rs:54:9
   |
54 |         pub type Assoc2 = (); //~ ERROR all associated items must have a visibility of `pub(crate)`
   |         ^^^

error: all associated items must have inherited visibility
  --> tests/ui/invalid.rs:61:9
   |
61 |         pub fn assoc2(&self) {} //~ ERROR all associated items must have inherited visibility
   |         ^^^
